src/code/utilities/genetics-utils.js

```diff
@@ -471,4 +471,59 @@ export default class GeneticsUtils {
     });
     return {geneStart, chromosomeHeight: species.chromosomesLength[chromosomeName]};
   }
+
+  /**
+   * Given a gene string, returns an object representing the phenotypes expressed by that gene string. For example,
+   * 'a:w,b:w' is converted to {'wings': 'No wings'}.
+   */
+  static convertGeneStringToPhenotype(genes) {
```

Comment:
This code got a little away from me, but I haven't spent time optimizing it, because I think specs are likely to change. This is an attempt to turn our basket definitions (which come in the form of allele strings) into what the ITS expects (i.e. accepted phenotypes).

For each basket, I end up passing over a mapping of the phenotypes they expect e.g. `{'wings': 'No wings'}` which I derive from the allele strings of the baskets. However, we're already realizing this is an insufficient way to describe the baskets - for example, it cannot describe a basket that accepts either 3 or 5 armor plates. Rob hacked a solution to this on his side, but if baskets get any more complex, I think it will be best to specify whatever information the ITS needs in the authoring doc instead of trying to derive it from allele strings here.

Happy to discuss this issue more, it's proven difficult to solve well.